cassandane/Cassandane/Cyrus/Sieve.pm

```diff
@@ -3958,7 +3891,7 @@ EOF
 
 sub test_imip_publish
     :needs_component_sieve :needs_component_httpd :min_version_3_7
```

Comment:
Maybe I don't understand the new logic, yet, but isn't `needs_component_httpd` unnecessary here and in the following imip_tests? and it it isn't, wouldn't `component_httpd` imply that `service_httpd` is required?

Reply:
This is addressed by the big XXX comment in the filter function, where the magic happens -- for the time being, if you `:want_service_http` (to have a http service started), you must also `:need_component_httpd` (to ensure the test is skipped if the httpd binary doesn't exist).

There isn't yet any mapping between the two, though I'd like there to be eventually.  The names are different (n.b. "http" vs "httpd") and inherently unrelated (:needs_x_y comes from cyr_buildinfo output, service names come from Cassandane::ServiceFactory), so even though this particular example seems obvious, the general case isn't.  So I want to see if and how we actually use this stuff before I go spending time on deeper infrastructure that we might not use.